DQM/L1TMonitor/python/L1TCaloLayer1Summary_cff.py

```diff
@@ -0,0 +1,109 @@
+
+from L1Trigger.L1TCaloLayer1.simCaloStage2Layer1Summary_cfi import simCaloStage2Layer1Summary as _simCaloStage2Layer1Summary
+cicadaEmulFromDigis = _simCaloStage2Layer1Summary.clone(caloLayer1Regions = cms.InputTag("caloLayer1Digis", ""))
+L1TReEmul.replace(simCaloStage2Layer1Summary, cicadaEmulFromDigis)
+
+# TwinMux
+stage2L1Trigger.toModify(simTwinMuxDigis,
+    RPC_Source         = 'rpcTwinMuxRawToDigi',
+    DTDigi_Source      = 'twinMuxStage2Digis:PhIn',
+    DTThetaDigi_Source = 'twinMuxStage2Digis:ThIn'
+)
+# BMTF
+stage2L1Trigger.toModify(simBmtfDigis,
+DTDigi_Source         = "simTwinMuxDigis",
+DTDigi_Theta_Source   = "bmtfDigis"
+)
+# KBMTF
+stage2L1Trigger.toModify(simKBmtfStubs,
+srcPhi       = 'simTwinMuxDigis',
+srcTheta     = 'bmtfDigis'
+)
+# OMTF
+stage2L1Trigger.toModify(simOmtfDigis,
+    srcRPC  = 'muonRPCDigis',
+    srcCSC  = 'csctfDigis',
+    srcDTPh = 'bmtfDigis',
+    srcDTTh = 'bmtfDigis'
+)
+# EMTF
+stage2L1Trigger.toModify(simEmtfDigis,
+    CSCInput = 'emtfStage2Digis',
+    RPCInput = 'muonRPCDigis'
+)
+# Calo Layer1
+stage2L1Trigger.toModify(simCaloStage2Layer1Digis,
+    ecalToken = 'ecalDigis:EcalTriggerPrimitives',
+    hcalToken = 'hcalDigis:'
+)
+
+(~stage2L1Trigger).toModify(simRctDigis,
+    ecalDigis = ['ecalDigis:EcalTriggerPrimitives'],
+    hcalDigis = ['hcalDigis:']
+)
+(~stage2L1Trigger).toModify(simRpcTriggerDigis, label = 'muonRPCDigis')
+
+# if not hasattr(process, 'L1TReEmulPath'):
+#     process.L1TReEmulPath = cms.Path(process.L1TReEmul)    
+#     process.schedule.append(process.L1TReEmulPath)
+
+stage2L1Trigger_2017.toModify(simOmtfDigis,
+    srcRPC   = 'omtfStage2Digis',
+    srcCSC   = 'omtfStage2Digis',
+    srcDTPh  = 'omtfStage2Digis',
+    srcDTTh  = 'omtfStage2Digis'
+)
+
+stage2L1Trigger.toModify(simEmtfDigis,
+    CSCInput  = cms.InputTag('emtfStage2Digis'),
+    RPCInput  = cms.InputTag('muonRPCDigis'),
+    CPPFInput = cms.InputTag('emtfStage2Digis'),
+    GEMEnable = cms.bool(False),
+    GEMInput  = cms.InputTag('muonGEMPadDigis'),
+    CPPFEnable = cms.bool(True), # Use CPPF-emulated clustered RPC hits from CPPF as the RPC hits
+)
+
+run3_GEM.toModify(simMuonGEMPadDigis,
+    InputCollection         = 'muonGEMDigis',
+)
+
+run3_GEM.toModify(simTwinMuxDigis,
+    RPC_Source         = 'rpcTwinMuxRawToDigi',
+    DTDigi_Source      = 'simDtTriggerPrimitiveDigis',
+    DTThetaDigi_Source = 'simDtTriggerPrimitiveDigis'
+)
+
+run3_GEM.toModify(simKBmtfStubs,
+    srcPhi   = 'bmtfDigis',
+    srcTheta = 'bmtfDigis'
+)
+
+run3_GEM.toModify(simBmtfDigis,
+    DTDigi_Source       = 'bmtfDigis',
+    DTDigi_Theta_Source = 'bmtfDigis'
+)
+
+from DQM.L1TMonitor.L1TCaloLayer1Summary_cfi import l1tCaloLayer1Summary as _l1tCaloLayer1Summary
+l1tCaloLayer1Summary = _l1tCaloLayer1Summary.clone(simCICADAScore = cms.InputTag("cicadaEmulFromDigis", "CICADAScore"))
+l1tCaloLayer1SummarySeq = cms.Sequence(L1TReEmul * l1tCaloLayer1Summary)
```

Comment:
@cms-sw/dqm-l2 This includes the entirety of L1T Re-emulation as part of this new sequence. That seems exdcessive to me. Was this approved at the DQM side?

Reply:
A lot of the code in this particular cff is admittedly boilerplate, since I was under the impression that cmsRun typically does lazy execution of producers. I will slim it down to only define the digis and run the modules that are necessary for the DQM module. This may take a couple days as I am not intimately familiar with the emulator sequences.

Reply:
In this case, it should just be the `simCaloStage2Layer1Summary` from here: https://github.com/cms-sw/cmssw/blob/master/L1Trigger/L1TCaloLayer1/python/simCaloStage2Layer1Summary_cfi.py if you are feeding it with unpacked CaloLayer1 Inputs.

Reply:
Dr. Savin requested plots that compare data to emulated regions, which necessitates the running of [L1TCaloLayer1](https://github.com/cms-sw/cmssw/blob/master/L1Trigger/L1TCaloLayer1/plugins/L1TCaloLayer1.cc). This module is not self contained and requires geometry information among other things, which are not included in the DQM sequence and so need to be added manually.